jujugui/static/gui/src/app/components/service-overview/service-overview.js

```diff
@@ -199,6 +199,31 @@ YUI.add('service-overview', function() {
       this.setState({confirmationOpen: false});
     },
 
+    /**
+      Handle destroying the service from the button click.
+
+      @method _destroyService
+    */
+    _destroyService: function() {
+      this._hideConfirmation();
+      // db, env, and service have already been bound to this function in
+      // the app.js definition.
+      this.props.destroyService(this._navigateToServices);
+      this.props.clearState();
```

Comment:
After all the debugging I feel like adding a comment here about why this is required would be nice to have.